From 5a78e6ae05ddf099d62e956c32b07ade7321b9b5 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?R=C3=A9mi=20Bernon?= <rbernon@codeweavers.com>
Date: Thu, 29 May 2025 10:44:32 +0200
Subject: [PATCH] frames/window-tracker: Remove unnecessary frame existence
 check

It's already checked inside remove_frame.

Part-of: <https://gitlab.gnome.org/GNOME/mutter/-/merge_requests/4465>
(cherry picked from commit dbd29639cb4a3192b5c0a895a2203d1f9498c4df)
---
 src/frames/meta-window-tracker.c | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/src/frames/meta-window-tracker.c b/src/frames/meta-window-tracker.c
index cfd62b6a58..247248d108 100644
--- a/src/frames/meta-window-tracker.c
+++ b/src/frames/meta-window-tracker.c
@@ -291,9 +291,7 @@ on_xevent (GdkDisplay *display,
           !g_hash_table_contains (window_tracker->client_windows,
                                   GUINT_TO_POINTER (xwindow)))
         set_up_frame (window_tracker, xwindow);
-      else if (xevent->xproperty.state == PropertyDelete &&
-               g_hash_table_contains (window_tracker->client_windows,
-                                      GUINT_TO_POINTER (xwindow)))
+      else if (xevent->xproperty.state == PropertyDelete)
         remove_frame (window_tracker, xwindow);
     }
   else if (xevent->type == PropertyNotify)
-- 
GitLab

